lib/pf/client/queue.pm

```diff
@@ -0,0 +1,141 @@
+package pf::client::queue;
```

Comment:
This should be a base class or at least not tied to redis so we can change the client

So this could be the base class 
Then pf::client::queue::redis could have the implementation.

Or anything else you find better, just not have the redis code in this class